.github/workflows/merge-to-main.js

```diff
@@ -0,0 +1,70 @@
+const {
+  slackNotification,
+  getLocalConfigs,
+  isWithinRCP,
+  pulls: { addLabels, addFiles, getChecks, getReviews },
+} = require('./helpers.js');
+
+// Run from the root of the project for local testing: node --env-file=.env .github/workflows/merge-to-main.js
+const PR_TITLE = '[Release] Stage to Main';
+const STAGE = 'stage';
+const PROD = 'main';
+
+let github, owner, repo;
+
+const getStageToMainPR = () =>
+  github.rest.pulls
+    .list({ owner, repo, state: 'open', base: PROD, head: STAGE })
+    .then(({ data } = {}) => data.find(({ title } = {}) => title === PR_TITLE))
+    .then((pr) => pr && addLabels({ pr, github, owner, repo }));
+
+const workingHours = () => {
+  const now = new Date();
+  const day = now.getUTCDay();
+  const hour = now.getUTCHours();
+  const isSunday = day === 0;
+  const isSaturday = day === 6;
+  const isFriday = day === 5;
+  return hour >= 8 && hour <= 20 && !isFriday && !isSaturday && !isSunday;
+};
+
+const main = async (params) => {
+  github = params.github;
+  owner = params.context.repo.owner;
+  repo = params.context.repo.repo;
+
+  if (isWithinRCP()) return console.log('Stopped, within RCP period.');
+  if (!workingHours()) return console.log('Stopped, outside working hours.');
+
+  try {
+    const stageToMainPR = await getStageToMainPR();
+    const signOffs = stageToMainPR?.labels.filter((l) => l.includes('SOT'));
+    console.log(`Currently ${signOffs.length} sign offs on the PR`);
+    if (signOffs.length >= 4) {
+      console.log('Stage to Main PR has all required labels. Merging...');
+      await github.rest.pulls.merge({
+        owner,
+        repo,
+        pull_number: stageToMainPR.number,
+        merge_method: 'merge',
+      });
```

Comment:
We will need to merge main into stage, after merging stage into main. Otherwise, our stage branch is always going to be outdated which might lead to issues trying to automatically merge a PR unless we force merge it. We don't want to force merge to avoid any potential issues.

Stage:
![Screenshot 2024-05-17 at 13 44 40](https://github.com/adobecom/milo/assets/39759830/efe1914b-c0a8-42b8-b889-ef536f65c161)

Main:
![Screenshot 2024-05-17 at 13 42 10](https://github.com/adobecom/milo/assets/39759830/6b23177d-8c2b-45db-ac61-f139391689d9)

Main receives an extra commit `e6d5f7c` due to the merge-commit. We will need to include `e6d5f7c` in stage as well, right after merging into main to get the two histories aligned.

Now as we'd merge `e6d5f7c` into stage, that would also always create a new commit, but this means stage is ahead of main, which is the normal case - so we'd still end up with an unnecessary commit but it's the lesser evil.

Reply:
Testing if that's really needed or an outdated branch doesn't stop the integration from merging.
I'd like to not back-merge if we can avoid it.

Reply:
Not needed. We can live with an outdated branch.